db/trie/mptrie/sortedlist.go

```diff
@@ -0,0 +1,58 @@
+package mptrie
+
+import (
+	"sort"
+)
+
+type sortedList struct {
+	li []uint8
+}
+
+func NewSortList(children map[byte]node) *sortedList {
+	if len(children) == 0 {
+		return &sortedList{
+			li: make([]uint8, 0),
+		}
+	}
+	li := make([]uint8, 0, len(children))
+	for k := range children {
+		li = append(li, k)
+	}
+	sort.Slice(li, func(i, j int) bool {
+		return li[i] < li[j]
+	})
+	return &sortedList{
+		li: li,
+	}
+}
+
+func (sl *sortedList) Insert(key uint8) {
+	i := sort.Search(len(sl.li), func(i int) bool {
+		return sl.li[i] >= uint8(key)
+	})
+	if i == len(sl.li) {
+		sl.li = append(sl.li, key)
+	} else {
+		if sl.li[i] != key {
```

Comment:
this should always be true? b/c the condition to call `Insert` is only key does not exist in children now
```
if _, exist := b.children[key]; !exist {
			b.indices.Insert(key)
		}
```

Reply:
also `make lint`, it may ask you to add comments for these funcs